cmd/api/protos/mood-api.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package mood;
+
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/doniacld/grpc-in-go/cmd/api/protos";
+
+// MoodType, the possible values for a mood.
+enum MoodState {
+  Happy = 0;
+  Sad = 1;
+}
+
+// Mood definition.
+message Mood {
+  MoodState State = 1;
+  // Date in epoch time.
+  int64 Date = 2;
```

Comment:
I believe, from what I gather of your requirements, that the Date should be separate from the rest. Maybe a Mood object that will be saved for a given Date will make a Log. Then you can add Habits to the Log object. And then you can add PagesRead and Achievements and whatever.